src/__tests__/helpers.js

```diff
@@ -0,0 +1,167 @@
+import {
+  guessMaxImageWidth,
+  bytesToSize,
+  selectSrc,
+  fallbackParams,
+} from '../components/helpers'
+
+describe('guessMaxImageWidth', () => {
+  beforeEach(() => {
+    Object.defineProperty(window.screen, 'width', {
+      value: 100,
+      writable: true,
+    })
+  })
+
+  it('Should calculate the maximum image width', () => {
+    const dimensions = {
+      width: 400,
+      height: 100,
+    }
+    const expected = dimensions.width
+    const result = guessMaxImageWidth(dimensions)
```

Comment:
Testing with global state is not always comfortable. You can pass window to this function, to make testing easier

e.g.

```js
const guessMaxImageWidth = (dimensions, {w = window}) {
  //use only w inside, but not window
}
```

later in test

```js
guessMaxImageWidth(dimensions, {w: fakeWindowObject})
```

Reply:
Can you help me understand how you say that `guessMaxImageWidth` can take window as a parameter? I don't see that anywhere in the source.

My understanding of the `guessMaxImageWidth` function is that it uses the global window to calculate (thus it's not a pure function). Because of this, I needed to mock the window dimensions to test the different branches of logic.

Reply:
I meant you can change function at your convenience to make it easier to test. You will be able to pass objects directly instead of assigning global variable

Reply:
Oh okay, makes sense! I will make those changes then 😄 

Had initially thought not to touch the source code until tests were in place and then to go back and revise things.